helper/validation/meta.go

```diff
@@ -66,8 +66,21 @@ func ToDiagFunc(validator schema.SchemaValidateFunc) schema.SchemaValidateDiagFu
 	return func(i interface{}, p cty.Path) diag.Diagnostics {
 		var diags diag.Diagnostics
 
-		attr := p[len(p)-1].(cty.GetAttrStep)
-		ws, es := validator(i, attr.Name)
+		// A practitioner-friendly key for any SchemaValidateFunc output.
+		// Generally this should be the last attribute name on the path.
+		// If not found for some unexpected reason, an empty string is fine
```

Comment:
Can you provide a bit of details about why it wouldn't not be, in some cases, that the last element in the path is not the one that implements `GetAttrStep`?
And if it's not, then what are those? `IndexStep`?

Reply:
FYI I dug a bit into `go-cty` to try to understand this a bit: I do get that those `*Step` structs are representation of operations that can be done on a `Path`, but I don't fully follow how they get assembled and why we would hit a scenario like the one described in the comment you added.

Reply:
> why it wouldn't not be, in some cases, that the last element in the path is not the one that implements GetAttrStep?

An `IndexStep` in `cty` represents a path inside a list, map, or set for one of the elements. For example, the path to the first element in a list is 0.

> I don't fully follow how they get assembled

Paths are assembled as part of decoding a value. The core folks would be best to ask how this works in terms of how a configuration/schema gets decoded to certain types and paths.

> why we would hit a scenario like the one described in the comment you added

If a new path type is added to cty. Just coding for future safety because the runtime code for providers should never introduce an unnecessary panic.

Reply:
I tried out the [`forcetypeassert` linter](https://github.com/gostaticanalysis/forcetypeassert) in golangci-lint and it catches some other potentially problematic code, but unfortunately doesn't catch this particular code pattern of a type assertion on a slice element. Not sure its worth the effort of enabling it in this project, but maybe some of the others.

Reply:
Big 👍 from me on adding this linter everywhere.